weld/subsystem/src/main/java/org/jboss/as/weld/util/PrivilegedActions.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat Inc., and individual contributors as indicated
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.weld.util;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedExceptionAction;
+
+/**
+ * Utils for working with {@link PrivilegedAction}.
+ *
+ * @author Jozef Hartinger
+ *
+ */
+public class PrivilegedActions {
```

Comment:
Do we really need this? It seems to be a class that exists only to remove a cast, and now that we are starting to really try and reduce our runtime memory footprint this seems like it may be unnecessary?

Reply:
It should remove the ambiguity between `PrivilegedAction` and `PrivilegedExceptionAction` which occurs when a lambda is passed to `AccessController`.

Reply:
Yes, but you can do that explicitly via a cast which will not add anything to class loading time or runtime memory usage:

 (AccessController.doPrivileged((PrivilegedAction<Integer>) () -> 1);)

Reply:
Ok, let's do it this way. On the other hand, such a small class does not affect the runtime memory usage significantly. I mean there are most probably better candidates in WildFly codebase and dependencies.